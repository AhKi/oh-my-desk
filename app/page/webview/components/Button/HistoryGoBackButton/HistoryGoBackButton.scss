.HistoryGoBackButton{
  display: flex;
  align-items: stretch;
  height: 100%;
}
