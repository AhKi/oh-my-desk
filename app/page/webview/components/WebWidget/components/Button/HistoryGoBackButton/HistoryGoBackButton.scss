.HistoryGoBackButton {
  display: inline;
}

.HistoryGoBackButton__btn {
  border: none;
  color: #F1F1F1;
  width: 24px;
  height: 28px;

  &:disabled {
    cursor: default;
    opacity: 0.5;
    &:hover {
      background-color: transparent;
      color: #F1F1F1;
    }
  }

  &:hover {
    background-color: #3F3F41;
    color: black;
  }
}
