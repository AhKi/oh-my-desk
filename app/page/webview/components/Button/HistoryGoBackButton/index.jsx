import React from 'react';
import PropTypes from 'prop-types';
import WidgetBackIcon from 'assets/icon/icon-widget-back-arrow';
import WidgetGoIcon from 'assets/icon/icon-widget-go-arrow';
import cx from 'classnames';
import './HistoryGoBackButton.scss';

const propTypes = {
  className: PropTypes.string,
  isCanGoBack: PropTypes.bool,
  isCanGoForward: PropTypes.bool,
  onGoBack: PropTypes.func,
  onGoForward: PropTypes.func,
};
const defaultProps = {
  className: '',
  isCanGoBack: false,
  isCanGoForward: false,
  onGoBack() {},
  onGoForward() {},
};

function HistoryGoBackButton(props) {
  const {
    className,
    isCanGoBack,
    isCanGoForward,
    onGoBack,
    onGoForward,
  } = props;
  return (
    <div
      className={cx('HistoryGoBackButton', className)}
    >
      <button
        type="button"
        className="WidgetHeader__button HistoryGoBackButton__btn--go-back"
        disabled={!isCanGoBack}
        onClick={onGoBack}
      >
        <WidgetBackIcon />
      </button>
      <button
        type="button"
        className="WidgetHeader__button HistoryGoBackButton__btn--go-front"
        disabled={!isCanGoForward}
        onClick={onGoForward}
      >
        <WidgetGoIcon />
      </button>
    </div>
  );
}

HistoryGoBackButton.propTypes = propTypes;
HistoryGoBackButton.defaultProps = defaultProps;

export default HistoryGoBackButton;
