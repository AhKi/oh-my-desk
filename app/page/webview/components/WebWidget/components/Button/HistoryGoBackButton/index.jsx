import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './HistoryGoBackButton.scss';

const propTypes = {
  className: PropTypes.string,
  isCanGoBack: PropTypes.bool,
  isCanGoForward: PropTypes.bool,
  onGoBack: PropTypes.func,
  onGoForward: PropTypes.func,
};
const defaultProps = {
  className: '',
  isCanGoBack: false,
  isCanGoForward: false,
  onGoBack() {},
  onGoForward() {},
};

function HistoryGoBackButton(props) {
  const {
    className,
    isCanGoBack,
    isCanGoForward,
    onGoBack,
    onGoForward,
  } = props;
  return (
    <div
      className={cx('HistoryGoBackButton', className)}
    >
      <button
        type="button"
        className="HistoryGoBackButton__btn HistoryGoBackButton__btn--go-back"
        disabled={!isCanGoBack}
        onClick={onGoBack}
      >
        <i className="fas fa-arrow-left" />
      </button>
      <button
        type="button"
        className="HistoryGoBackButton__btn HistoryGoBackButton__btn--go-front"
        disabled={!isCanGoForward}
        onClick={onGoForward}
      >
        <i className="fas fa-arrow-right" />
      </button>
    </div>
  );
}

HistoryGoBackButton.propTypes = propTypes;
HistoryGoBackButton.defaultProps = defaultProps;

export default HistoryGoBackButton;
