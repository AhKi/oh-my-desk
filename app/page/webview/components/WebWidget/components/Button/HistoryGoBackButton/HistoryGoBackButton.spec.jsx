import React from 'react';
import { shallow } from 'enzyme';

import HistoryGoBackButton from './';

describe('<HistoryGoBackButton />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<HistoryGoBackButton />);

    expect(wrapper).toMatchSnapshot();
  });
});
