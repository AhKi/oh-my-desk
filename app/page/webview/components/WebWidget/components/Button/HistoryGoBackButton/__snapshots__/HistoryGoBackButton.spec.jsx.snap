// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryGoBackButton /> should match to snapshot when render default 1`] = `
<div
  className="HistoryGoBackButton"
>
  <button
    className="HistoryGoBackButton__btn HistoryGoBackButton__btn--go-back"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-arrow-left"
    />
  </button>
  <button
    className="HistoryGoBackButton__btn HistoryGoBackButton__btn--go-front"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-arrow-right"
    />
  </button>
</div>
`;
